src/lib.rs

```diff
@@ -146,6 +146,11 @@ pub mod errors {
                 description("A SUMMARY.md parsing error")
                 display("Error at line {}, column {}: {}", line, col, message)
             }
+
+            ReservedFilenameError(message: String, filename: String) {
```

Comment:
You can probably simplify this by removing the `message` parameter. If it's a reserved filename then the message is almost always going to be "{} is reserved for internal use".

i.e. something like this:

```rust
ReservedFilenameError(filename: PathBuf) {
  description("Reserved Filename")
  display("{} is reserved for internal use", filename.display())
}
```